csharp/src/Apache.Arrow/Memory/LargeNativeMemoryAllocator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal class LargeNativeMemoryAllocator:LargeMemoryAllocator
```

Comment:
As mentioned on the `LargeMemoryAllocator`, this logic should be added to the existing `NativeMemoryAllocator`  class rather than be a separate class.